Automattic/jetpack
# Add a 'Check for spam' button on the new feedback dashboard

## Proposed changes:

This patch adds a 'Check for spam' button to bulk actions on the new feedback dashboard. In order to keep the functionality of the existing implementation, the button works slightly differently though, as was discussed with @digitalwaveride:

- The button will only show when you're on the 'Inbox' tab and you have at least one response selected.
- Once you click the button, it'll start processing the entire queue to check for spam (regardless of your selection) in batches of 100 responses at a time. A loading spinner will show.
- Once it's finished processing feedback responses, the loading spinner will disappear and your results will refresh.

![Screenshot 2023-04-06 at 12 39 57](https://user-images.githubusercontent.com/8056203/230354954-2195c9de-eec4-472e-8b74-316c4a029306.png)

### Other information:


## Jetpack product discussion



pcsBup-Pl-p2

## Does this pull request change what data or activity we track or use?



No.

## Testing instructions:






- Go to `admin.php?page=jetpack-forms`.
- Select at least one feedback response.
- You should see the check for spam button.
- Clicking the check for spam should trigger the check and you should see the loading indicator on the button.
- Once processing spam has finished, the results should refresh.
- You should not see the check for spam button on either 'Spam' or 'Trash' tabs.

_Tip:_ If you don't have over 100 responses, you can adjust the limit on line `18` in `bulk-actions-menu.js` to something lower to see the requests queueing up correctly.
